### PR TITLE
KafkaChannel validation tweak

### PR DESCRIPTION
**Pull request by matzew**
Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

Fixes #

## Proposed Changes

- relaxing the validation, like on the source and others in eventing

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: kafka/channel/config/300-kafka-channel.yaml [hunk 1]
```diff
@@ -23,7 +23,15 @@ metadata:
     duck.knative.dev/addressable: "true"
 spec:
   group: messaging.knative.dev
-  version: v1alpha1
+  preserveUnknownFields: false
```
Comment (devguyio) [inline kafka/channel/config/300-kafka-channel.yaml:26]: from my understanding from [here](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#preserving-unknown-fields) this is like "enable pruning" , and then in line 34 you're saying "disable pruning" . Is there a reason for that?

Comment (matzew) [reply kafka/channel/config/300-kafka-channel.yaml:26]: following like all others did it, when going this route.

One example: https://github.com/knative/eventing/blob/release-0.15/config/core/resources/containersource.yaml#L37-L45

**Pull request merged**